sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1515,13 +1554,29 @@ def task_worker(self):
                             self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_FAILED
                             continue
                         host_lanes_mask = self.port_dict[lport]['host_lanes_mask']
-                        self.log_notice("{}: Setting lanemask=0x{:x}".format(lport, host_lanes_mask))
+                        self.log_notice("{}: Setting host_lanemask=0x{:x}".format(lport, host_lanes_mask))
+			
+			appl_advt = api.get_application_advertisement()
+                        appl_code = appl_advt[appl]
+                        self.port_dict[lport]['media_lane_count'] = appl_code['media_lane_count']
```

Comment:
Can we modify get_media_lane_count and get_media_lane_assignment_option APIs to return o/p based on the selected application?

Reply:
sure.

Reply:
Updated the api's to retrieve the media_lane_count and media_lane_assignment_options and raised the PR using 
https://github.com/sonic-net/sonic-platform-common/pull/368